cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/AbstractConnectOperator.java

```diff
@@ -0,0 +1,346 @@
+        this.imagePullPolicy = config.getImagePullPolicy();
+        this.imagePullSecrets = config.getImagePullSecrets();
+        this.operationTimeoutMs = config.getOperationTimeoutMs();
+        this.pfa = pfa;
+    }
+
+    @Override
+    protected Future<Boolean> delete(Reconciliation reconciliation) {
+        // When deleting KafkaConnect we need to update the status of all selected KafkaConnector
+        return connectorOperator.listAsync(reconciliation.namespace(), Labels.forCluster(reconciliation.name())).compose(connectors -> {
+            List<Future> connectorFutures = new ArrayList<>();
+            for (KafkaConnector connector : connectors) {
+                connectorFutures.add(maybeUpdateConnectorStatus(reconciliation, connector,
+                        noConnectCluster(reconciliation.namespace(), reconciliation.name())));
+            }
+            return CompositeFuture.join(connectorFutures);
+        }).map(ignored -> Boolean.FALSE);
+    }
+
+    /**
+     * Create a watch on {@code KafkaConnector} in the given {@code namespace}.
+     * The watcher will:
+     * <ul>
+     * <li>{@link #reconcileConnectors(Reconciliation, CustomResource)} on the KafkaConnect or KafkaConnectS2I
+     * identified by {@code KafkaConnector.metadata.labels[strimzi.io/cluster]}.</li>
+     * <li>If there is a Connect and ConnectS2I cluster with the given name then the plain Connect one is used
+     * (and an error is logged about the ambiguity).</li>
+     * <li>The {@code KafkaConnector} status is updated with the result.</li>
+     * </ul>
+     * @param connectOperator The operator for {@code KafkaConnect}.
+     * @param connectS2IOperator The operator for {@code KafkaConnectS2I}.
+     * @param namespace The namespace.
+     * @return A future which completes when the watch has been set up.
+     */
+    public static Future<Void> createConnectorWatch(AbstractConnectOperator<KubernetesClient, KafkaConnect, KafkaConnectList, DoneableKafkaConnect, Resource<KafkaConnect, DoneableKafkaConnect>> connectOperator,
+            AbstractConnectOperator<OpenShiftClient, KafkaConnectS2I, KafkaConnectS2IList, DoneableKafkaConnectS2I, Resource<KafkaConnectS2I, DoneableKafkaConnectS2I>> connectS2IOperator,
+            String namespace) {
+        // TODO Only create a single watch
+        return Util.async(connectOperator.vertx, () -> {
+            connectOperator.connectorOperator.watch(namespace, new Watcher<KafkaConnector>() {
+                @Override
+                public void eventReceived(Action action, KafkaConnector kafkaConnector) {
+                    String connectName = kafkaConnector.getMetadata().getLabels().get(Labels.STRIMZI_CLUSTER_LABEL);
+                    Future<Void> f;
+                    if (connectName != null) {
+                        // Check whether a KafkaConnect/S2I exists
+                        CompositeFuture.join(connectOperator.resourceOperator.getAsync(namespace, connectName),
```

Comment:
What kind of namespace is the `namespace` here? I think every code in this section needs to use the namespace of the KafkaConnector resource. Otherwise how do you handle watching multiple / all namespaces?

Reply:
Good catch. Really it points out the hackery of using the string "*". We should seriously consider using `Optional` (or some other class than String) distinguish the all namespaces case from the single namespace case. Because it's just too easy lose track of which Strings are definitely namespaces and which are "namespace patterns/wildcards". But that's for an other PR.